proof/crefine/RISCV64/Arch_C.thy

```diff
@@ -1595,30 +1661,40 @@ lemma decodeRISCVFrameInvocation_ccorres:
      apply (rule ccorres_rhs_assoc)+
      apply (ctac add: setThreadState_ccorres)
        apply csymbr
+       apply (rule ccorres_nondet_refinement)
+        apply (rule is_nondet_refinement_bindE)
+         apply (rule is_nondet_refinement_refl)
+        apply (simp split: if_split)
+        apply (rule conjI[rotated], rule impI, rule is_nondet_refinement_refl)
+        apply (rule impI)
+        apply (rule is_nondet_refinement_alternative1)
```

Comment:
this chain looks like something `fastforce` or `auto` could solve given `is_nondet_refinement_refl` and `is_nondet_refinement_alternative1` as `intro` or `intro!` rules.

Reply:
Would be nice to try out yes. The `conjI[rotated]` indicates that there is something going on here, possibly an instantiation that fastforce/auto might not get. But the lines after that could be collapsed.